Framework/Algorithms/test/MonteCarloAbsorptionTest.h

```diff
@@ -52,11 +52,13 @@ void addSample(Mantid::API::MatrixWorkspace_sptr ws,
     const V3D axis(0.0, 1.0, 0.0);
 
     ShapeFactory shapeMaker;
-    auto can = boost::make_shared<Container>(
+    auto canShape =
         shapeMaker.createShape(ComponentCreationHelper::cappedCylinderXML(
-            radius, height, baseCentre, axis, id)));
-    can->setMaterial(Material("CanMaterial",
-                              PhysicalConstants::getNeutronAtom(26, 0), 0.01));
+            radius, height, baseCentre, axis, id));
+    // Set material assuming it's a CSG Object
```

Comment:
I don't think you need this comment. There is no assumption. `createShape` will only return a CSGObject